consensus/src/dag/reconstruction/mod.rs

```diff
@@ -25,7 +25,17 @@ impl<U: Unit> ReconstructedUnit<U> {
         match unit.control_hash().combined_hash
             == ControlHash::<U::Hasher>::combine_hashes(&parents)
         {
-            true => Ok(ReconstructedUnit { unit, parents }),
+            true => {
+                let unit_round = unit.round();
+                let mut parents_with_rounds = NodeMap::with_size(parents.size());
+                for (parent_index, hash) in parents.into_iter() {
+                    parents_with_rounds.insert(parent_index, (hash, unit_round - 1));
```

Comment:
This can underflow, but in practice `unit_round` is always positive here. Add `assert`?

Reply:
I would go with a `saturating_sub` plus comment that it doesn't matter. Or maybe not even that, since this code is extremely temporary and likely will never even get released, never mind encountering any strange states. :P